### PR TITLE
de-virtualize elfobj.c and machobj.c

### PR DESCRIPTION
**Pull request by WalterBright**
A step towards converting elfobj.c and machobj.d to D. It removes the ElfObj, MachObj, and MachObj64 classes.

### CODE REVIEW

File: src/dmd/backend/obj.h [hunk 1]
```diff
@@ -156,8 +156,8 @@ class Obj
     VIRTUAL int  comdatsize(Symbol *, targ_size_t symsize);
     virtual int readonly_comdat(Symbol *s);
```
Comment (rainers) [inline src/dmd/backend/obj.h]: Now static on the D side, too.

**Comment by jacob-carlborg** [comment]
How far are we going to take this devirtulization? I think the code looks much better with the virtual methods instead of having if statements all over the place?

**Comment by jacob-carlborg** [comment]
> The only time virtuals even have a point is for Windows. The rest should be static.

It had a point for MachObj.

**Comment by thewilsonator** [comment]
> It had a point for MachObj.

That was 32 vs 64 bit right? Why do we even support 32bit OSX anymore?

**Comment by jacob-carlborg** [comment]
> That was 32 vs 64 bit right?

Yes.

> Why do we even support 32bit OSX anymore?

I think @p0nce is still using 32bit for his plugins. Not sure if LDC would be enough for him. I think Brad have complained in the past. Any features from me will be 64bit only, like new Objective-C features.

**Comment by WalterBright** [comment]
Debating supporting 32 bit OSX is off topic for this PR.

**Review by jacob-carlborg — APPROVED** [review]

**Pull request merged**